### ISSUE TITLE
Explore using different text distances to produce groupings

### ISSUE THREAD

**Issue by stuartlynn** [tsdataclinic/scout#179]
Try using a lot of different text distance metrics to see which gives the best results for clustering

**Comment by stuartlynn** [comment]
I attached the output of some column matching that I ran. After counting the frequencies of all column names, I took the top 200 columns and tried to match it against all the other columns, then picked an arbitrary similarity score threshold of >65% match. The csv files attached are the resulting matches, ordered by score.

There are two results using two distances metrics.
- levenshein_ratio_matches.csv: just uses levenshtein distance to calculate a score (hence no 100% match)
- token_sort_ratio_matches: tokenizes and sorts the string before comparison – hence solves “out of order” issues

Some brief observations:
- Plain levenshtein distance is good for capturing obvious cases where the columns are referring to the same thing, and good for catching typos
- Token_sort_ratio is good for catching out-of-order cases eg. “% student” vs “student %” will yield 100% match

We will need some form of human input and discretionary decision making to identify columns referring to the same entity. I was also working on some code to display the names of the datasets that the column names appear in. Maybe we could have an app that presents likely matches and have a human confirm it, and on ambiguous cases, present more information eg. which datasets contain said column name, and maybe present some column values too if the user selects a dataset to look at.
